spec/calculator_spec.rb

```diff
@@ -1 +1,95 @@
-# Your code here
\ No newline at end of file
+RSpec.describe Calculator do
+  let(:calc) { Calculator.new }
+
+  describe '#add' do
+    it 'adds two numbers' do
+      expect(calc.add(3, 5)).to eq 8
+      expect(calc.add(3, -5)).to eq -2
+      expect(calc.add(3.0, 5.5)).to eq 8.5
+    end
+  end
+
+  describe '#subtract' do
+    it 'subtracts two numbers' do
+      expect(calc.subtract(2, 1)).to eq 1
+      expect(calc.subtract(2.0, 1.0)).to eq 1
+      expect(calc.subtract(1, 2.5)).to eq -1.5
+      expect(calc.subtract(1, -2)).to eq 3
+    end
+  end
+
+  describe '#multiply' do
+    it 'multiplies two numbers' do
+      expect(calc.multiply(2, 1)).to eq 2
+      expect(calc.multiply(2.5, 3)).to eq 7.5
+      expect(calc.multiply(7, 6)).to eq 42
+      expect(calc.multiply(7, -6)).to eq -42
+    end
+  end
+
+  describe '#divide' do
+    it 'divides two numbers' do
+      expect(calc.divide(42, 7)).to eq 6
+      expect(calc.divide(12.0, 3)).to eq 4
+      expect(calc.divide(12, 7)).to be_within(0.001).of(1.7142)
+    end
+    context 'divide by zero' do
+      it 'raises an error' do
+        expect { calc.divide(1, 0) }.to raise_error ArgumentError
+        expect { calc.divide(1, 0.000) }.to raise_error ArgumentError
+      end
+    end
+  end
+
+  describe '#pow' do
+    it 'raises first number to the power of the second' do
+      expect(calc.pow(2, 3)).to eq 8
+      expect(calc.pow(2, 1.233)).to be_within(0.0001).of(2.3505)
+      expect(calc.pow(2, -1)).to eq 0.5
+    end
+  end
+
+  describe 'sqrt' do
+    it 'returns the square root of a number' do
+      expect(calc.sqrt(9)).to eq 3
+      expect(calc.sqrt(8)).to eq 2.83
```

Comment:
the result for square root of a non-round argument should have 2 decimal places exactly